test/object-store/sync/flx_sync.cpp

```diff
@@ -42,16 +43,23 @@ class FLXSyncTestHarness {
     FLXSyncTestHarness(const std::string& test_name, ServerSchema server_schema = default_server_schema());
 
     template <typename Func>
-    void do_with_new_realm(Func&& func, util::Optional<Schema> schema_for_realm = util::none)
+    void do_with_new_user(Func&& func)
```

Comment:
What's the advantage of making these take a callback rather than just returning the user?